test/geos_capi/tc_polygon.rb

```diff
@@ -79,6 +49,62 @@ def test_union
         end
 
 
+        def test_simplify
+          xys = [[0, 0], [5, 0], [10, 0], [10, 10], [5, 10.2], [0, 10], [0, 0]]
+          points = xys.collect { |x,y| @factory.point(x, y) }
+          poly = @factory.polygon(@factory.linear_ring(points))
+          simplified = poly.simplify(0.3)
+          new_points = simplified.exterior_ring.points
+          extra = new_points.reject { |p| [0, 10].include?(p.x) and [0, 10].include?(p.y) }
+          assert_equal 5, new_points.length, "Closed ring of the square should have 5 points"
+          assert_equal 0, extra.length , "Should only have x/y's on 0 and 10"
+        end
+
+        def test_buffer
+          polygon_coordinates = [[0.5527864045000421, 3.776393202250021],
+                                 [0.7763932022500211, 4.447213595499958],
+                                 [1.4472135954999579, 4.223606797749979],
+                                 [2.447213595499958, 2.223606797749979],
+                                 [2.223606797749979, 1.5527864045000421],
+                                 [1.5527864045000421, 1.776393202250021],
+                                 [0.5527864045000421, 3.776393202250021]]
+
+          points_arr = polygon_coordinates.map{|v| @factory.point(v[0],v[1])}
```

Comment:
Space missing after comma.<br>Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.